pom.xml

```diff
@@ -129,6 +129,60 @@ Contributors:
                 <test.exclusion.pattern>nothing</test.exclusion.pattern>
             </properties>
         </profile>
+        <profile>
+            <id>jdk8orGreater</id>
+            <activation>
+                <jdk>[1.8,)</jdk>
+            </activation>
+            <properties>
+                <jdk8orGreater>true</jdk8orGreater>
+            </properties>
+        </profile>
+        <profile>
+            <!-- This profile activate the usage of toolchain to compile using JDK 1.7-->
+            <id>useToolchain</id>
+            <properties>
```

Comment:
is the `<activation>` left out by intention?

Reply:
This profile is intended to be activated manually.

Reply:
I use [Java8Test](https://github.com/eclipse/californium/blob/check_java8/element-connector/src/test/java/org/eclipse/californium/elements/Java8Test.java) to check, if java 7 is really used.

Reply:
OK, I see ...

But I guess, you use the keytool of java newer than java 8. The keystore type jks is the default up to java 8 and afterwards changed to PKCS12, which is not supported by java 7. I added therefore the keystore type explicitly to the script.

Reply:
Looking at [Jenkins ouput](https://ci.eclipse.org/leshan/job/leshan/709/console) : 
```
[INFO] ------------< org.eclipse.leshan:leshan-integration-tests >-------------
[INFO] Building leshan - integration tests 2.0.0-SNAPSHOT                [9/12]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ leshan-integration-tests ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven) @ leshan-integration-tests ---
[INFO] 
[INFO] --- maven-toolchains-plugin:3.0.0:toolchain (default) @ leshan-integration-tests ---
[INFO] Required toolchain: jdk [ version='1.7' ]
[INFO] Found matching toolchain for type jdk: JDK[/opt/tools/java/oracle/jdk-7/latest]
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ leshan-integration-tests ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/jenkins/agent/workspace/leshan/leshan-integration-tests/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ leshan-integration-tests ---
[INFO] Toolchain in maven-compiler-plugin: JDK[/opt/tools/java/oracle/jdk-7/latest]
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ leshan-integration-tests ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/jenkins/agent/workspace/leshan/leshan-integration-tests/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ leshan-integration-tests ---
[INFO] Toolchain in maven-compiler-plugin: JDK[/opt/tools/java/oracle/jdk-7/latest]
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 39 source files to /home/jenkins/agent/workspace/leshan/leshan-integration-tests/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M4:test (default-test) @ leshan-integration-tests ---
[INFO] Toolchain in maven-surefire-plugin: JDK[/opt/tools/java/oracle/jdk-7/latest]
```

So java7 seems correctly used. 

But regarding  https://github.com/eclipse/californium/pull/1442#issuecomment-724853567, I try to build #923 and it failed with : 
```
[ERROR] org.eclipse.leshan.integration.tests.BootstrapTest.bootstrapSecureWithBadPSKKey  Time elapsed: 0.001 s  <<< ERROR!
java.lang.RuntimeException: java.io.IOException: Invalid keystore format
	at org.eclipse.leshan.integration.tests.BootstrapTest.<init>(BootstrapTest.java:47)
Caused by: java.io.IOException: Invalid keystore format
	at org.eclipse.leshan.integration.tests.BootstrapTest.<init>(BootstrapTest.java:47)

[ERROR] org.eclipse.leshan.integration.tests.BootstrapTest.bootstrapWithDiscoverOnDevice  Time elapsed: 0 s  <<< ERROR!
java.lang.RuntimeException: java.io.IOException: Invalid keystore format
	at org.eclipse.leshan.integration.tests.BootstrapTest.<init>(BootstrapTest.java:47)
Caused by: java.io.IOException: Invalid keystore format
	at org.eclipse.leshan.integration.tests.BootstrapTest.<init>(BootstrapTest.java:47)
```
This confirms that java7 is used and you are right about keystore format.
(Thx again for letting me know that :pray: )